### ISSUE TITLE
diff() along axis of size 1

### ISSUE THREAD

**Issue by ben-bou** [helmholtz-analytics/heat#753]
**Description**
A clear and concise description of the bug and the associated functionality.

The edge case of performing ht.diff(array) along an axis of size 1 is handled differently when array is distributed.

**To Reproduce**
1. Which module/class/function is affected?
ht.diff()
2. What are the circumstances under which the bug appears?
array is distributed
3. What is the exact error message / erroneous behavior?
```
Traceback (most recent call last):
  File "tmp_test.py", line 4, in <module>
    print(ht.diff(arr, axis=1).shape)
  File "/p/.../heat/core/arithmetics.py", line 396, in diff
    ret.lloc[cr_slice].shape, dtype=ret.dtype.torch_type(), device=a.device.torch_device
  File "/p/.../heat/core/dndarray.py", line 48, in __getitem__
    return self.obj[key]
IndexError: index 1 is out of bounds for dimension 1 with size 1
```

**Expected behavior**
When the array is not distributed, no error occurs.
The code below results in a shape of (2, 0, 2).

**Illustrative**
```python
import heat as ht

arr = ht.ones((2,1,2), split=0)
print(ht.diff(arr, axis=1).shape)
```

**Version Info**
master branch

**Additional comments**
The error is in line 394 (https://github.com/helmholtz-analytics/heat/blob/master/heat/core/arithmetics.py#L394):
```python
cr_slice[axis] = 1
```
The index 1 does not exist in this edge case. Because this index is only used for determining the size of the buffer (and nothing else), it does not need to be 1. Changing the line to 
```python
cr_slice[axis] = slice(0, 1, None)
```
resolves the issue and passes the tests.
Thank you

**Comment by ClaudiaComito** [comment]
I'm ashamed to say this is still open. @mtar @JuanPedroGHM can one of you fix this? @ben-bou suggests a fix in his original post. Thanks!

**Comment by ClaudiaComito** [comment]
Closed by #1201
